woocommerce/woocommerce-gutenberg-products-block
# Finish wiring up shipping data context provider and add disabled state to checkout submit button

Fixes: #1944 

This pull finishes off the `ShippingDataProvider` component and implements the following:

- Adds selecting rate resolution state to the cart store and exposes on `useSelectShippingRate` hook.
- Refactors `ShippingDataProvider` so that:
    - remove un-needed component and move shipping rates into context provider (via `useShippingRates`).
    - remove error handling (currently the ui/ux for the checkout should result in no errors from ui/ux and any api errors are non-recoverable - thus thrown from the hooks). This may need revisited as we test but for now I'm removing the code that does nothing. 
    - Increment/decrement checkout calculation counts when shipping rates are being selected (shipping rate retrieval is already doing this).
- dispatch checkout error state update if there is a shipping rate error.
- disable the place order button component if checkout is calculating or checkout has an error.

## To Test

- This impacts the cart because of the hook changes. So verify that existing behaviour for shipping related ui/ux in the cart remains unaffected.
- This impacts anything to do with shipping in the Checkout block. So verify that behaviour for shipping related calculations and total updates continue to work as expected.
- Also verify in the Checkout block that when shipping rates are being changed, or selected that the place order button is disabled (and enables again after network request is done).
